src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsReconciliationService.kt

```diff
@@ -0,0 +1,78 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.incidents
+
+import com.microsoft.applicationinsights.TelemetryClient
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.withContext
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.helpers.trackEvent
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.IncidentAgencyId
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.doApiCallWithRetries
+
+@Service
+class IncidentsReconciliationService(
+  private val telemetryClient: TelemetryClient,
+  private val incidentsService: IncidentsService,
+  private val nomisIncidentsApiService: IncidentsNomisApiService,
+) {
+  private companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  suspend fun generateReconciliationReport(agencies: List<IncidentAgencyId>): List<MismatchIncidents> =
```

Comment:
Wasn't sure if the List of incidentAgencyId should be paged...there will be just over 120 ish returned

Reply:
nah, not worth it